zerver/worker/queue_processors.py

```diff
@@ -527,3 +535,37 @@ def consume(self, event: Mapping[str, Any]) -> None:
                 (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id,
                                                                require_active=False)
                 do_mark_stream_messages_as_read(user_profile, stream)
+
+
+@assign_queue('export_data')
+class DataExporter(QueueProcessingWorker):
+    def consume(self, email: str) -> None:
+        output_dir = tempfile.mkdtemp(prefix="/tmp/zulip-export-")
+        if os.path.exists(output_dir):
+            shutil.rmtree(output_dir)
```

Comment:
We can probably just assert that it doesn't exist; that should never happen given that `mkdtemp` is supposed to make us a secure, randomly generated path.  Also, does `mkdtemp` create the directory, in which case we should be asserting it does exist?

Reply:
Yes, this code is something I just copied over from the export command, and your suggestions make sense. :+1: